mozilla/fx-private-relay
# MPP-3119: complaint notification disables mask

# Disable mask on complaint notification
See https://mozilla-hub.atlassian.net/browse/MPP-3119 for more context.

When a user (or their email provider) marks a Relay email as spam, we receive a complaint notification. To prevent further spam from reaching the user (causing more complaints), we want to disable the email mask that forwarded the spam email, and notify the user.


# Screenshot (if applicable)
![image](https://github.com/user-attachments/assets/531c8e16-f11d-4db6-9647-417d9e2b73da)


# How to test
This will be hard to test until it's on a server with fully operational AWS SES. When it is:

0. Sign up for Relay with an FXA that has a gmail, yahoo, or outlook primary email address
1. In the django admin site, activate the new `disable_mask_on_complaint` flag for the user
2. Create an email mask
3. Send an email to that mask
4. In the final destination mailbox, mark that email as spam
   * Note: It's unclear if all email providers send these complaint notifications, but gmail, yahoo, outlook, and other popular providers seem to do so.
   * Note 2: It's also unclear how long it takes for AWS to process complaint notifications and send them to Relay
5. At some point, Relay should receive the complaint notification
   * [ ] Verify the mask is disabled
   * [ ] Verify the user received an email notification that their mask was disabled

# Checklist (Definition of Done)
